### PR TITLE
Task: Add optional parameter to support validation

### PR DESCRIPTION
**Pull request by thewahome**
Depends on the merging of [this PR](https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/pull/944) .

The new generator will need to validate the individual "platform" yaml files generated. It passes a new parameter "platformName" that tells the validation script which file to validate.

### Changes
This script will change from:
> ./scripts/run-openapi-validation.ps1 -repoDirectory C:/github/msgraph-metadata -version "v1.0" 

To
> ./scripts/run-openapi-validation.ps1 -repoDirectory C:/github/msgraph-metadata -version "v1.0" -platformName "openapi"

### Notes
To make sure things do not break, the `platformName` parameter is optional and set to the known available platform -> `openapi` to make sure things do not break

### CODE REVIEW

File: scripts/run-openapi-validation.ps1 [hunk 1]
```diff
@@ -16,15 +16,25 @@
...
 
 param(
     [Parameter(Mandatory=$true)][string]$repoDirectory,
-    [Parameter(Mandatory=$true)][string]$version
+    [Parameter(Mandatory=$true)][string]$version,
+    [Parameter(Mandatory=$false)][string]$platformName
 )
-$yaml = Join-Path $repoDirectory "openapi" $version "openapi.yaml"
+
+if([string]::IsNullOrWhiteSpace($aString))
```
Comment (baywet) [inline scripts/run-openapi-validation.ps1:30]: where is aString supposed to come from?

**Review by baywet — CHANGES_REQUESTED** [review]
we probably need the convert settings for that step as well.
https://github.com/microsoftgraph/msgraph-metadata/blob/c67be8e3838be1d26e474fe060d269646833930d/scripts/run-metadata-validation.ps1#L39

File: scripts/run-openapi-validation.ps1 (hunk 1 above)
Comment (thewahome) [reply scripts/run-openapi-validation.ps1:30]: Lemme fix that. It's supposed to be platformName

**Comment by thewahome** [comment]
> we probably need the convert settings for that step as well.
> https://github.com/microsoftgraph/msgraph-metadata/blob/c67be8e3838be1d26e474fe060d269646833930d/scripts/run-metadata-validation.ps1#L39

I probably have low context on what this file does however it wasn't run by the pipeline step that generates openapi files hence it's not required to be specific to any platform

**Comment by thewahome** [comment]
> we probably need the convert settings for that step as well.
> https://github.com/microsoftgraph/msgraph-metadata/blob/c67be8e3838be1d26e474fe060d269646833930d/scripts/run-metadata-validation.ps1#L39

I probably have low context on what this file does however it wasn't run by the pipeline step that generates openapi files hence it's not required to be specific to any platform

**Comment by baywet** [comment]
> I probably have low context on what this file does however it wasn't run by the pipeline step that generates openapi files hence it's not required to be specific to any platform

It validates changes made to the XSL /transformation rules before they are committed to the repo. That's to ensure any change to the XSL is working for the CSDL and not breaking the OpenAPI conversion. And since we're planning on changing the defaults in hidi, that validation will be impacted and should load settings. Ideally it'd actually validate against all settings using a matrix to make sure we smoke test for all of our experiences.

**Comment by thewahome** [comment]
>  And since we're planning on changing the defaults in hidi, that validation will be impacted and should load settings. Ideally it'd actually validate against all settings using a matrix to make sure we smoke test for all of our experiences.

@baywet there are no longer any changes needed on Hidi. I am adding a matrix to smoke test this

File: scripts/run-openapi-validation.ps1 (hunk 1 above)
Comment (baywet) [reply scripts/run-openapi-validation.ps1:30]:
```suggestion
if([string]::IsNullOrWhiteSpace($platformName))
```

File: .github/workflows/metadata-parser-validation.yml [hunk 2]
```diff
@@ -7,21 +7,39 @@ on:
...
       uses: actions/setup-dotnet@v1
       with:
         dotnet-version: 7.0.x
+
     - name: Validate latest XSLT rules
-      run: ./scripts/run-metadata-validation.ps1 -repoDirectory "${{ github.workspace }}" -version "${{ matrix.version }}"
+      run: ./scripts/run-metadata-validation.ps1 -repoDirectory "${{ github.workspace }}" -version "${{ matrix.version }}" -platformName "${{ matrix.settings }}"
       shell: pwsh
+
+    - uses: actions/upload-artifact@v3
```
Comment (baywet) [inline .github/workflows/metadata-parser-validation.yml:42]: very good initiative to upload things as artifacts!
```suggestion
    - uses: actions/upload-artifact@v3
      if: always()
```

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for moving things around. couple of minor comments

**Comment by baywet** [comment]
@thewahome can you double check the matrix aspect, it doesn't seem to be triggering the build jobs anymore. Thanks!

**Comment by thewahome** [comment]
> @thewahome can you double check the matrix aspect, it doesn't seem to be triggering the build jobs anymore. Thanks!

On it @baywet

**Comment by thewahome** [comment]
@baywet this is done.
The validations should fail until when the [SDK generator](https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/) is triggered to create the files

**Comment by baywet** [comment]
@thewahome could we add placeholders by just copy/pasting the existing ones just so we can make sure the pipeline changes work?

**Comment by thewahome** [comment]
> @thewahome could we add placeholders by just copy/pasting the existing ones just so we can make sure the pipeline changes work?

The key things to check are:
- whether the pipeline spawns the combinations of version and platform validation jobs.
- whether the spawned job checks in the correct folder location where the expected files will be placed.

We can see the spawned jobs. And on checking the one that failed we can see the error message shows that it went looking in the right place

> Could not validate the document, reason: 
> Could not open the file at /home/runner/work/msgraph-metadata/msgraph-metadata/openapi/v1.0/graphexplorer.yaml

**Review by baywet — APPROVED** [review]

**Pull request merged**